src/main/kotlin/com/openlattice/organizations/DatasetController.kt

```diff
@@ -108,7 +127,13 @@ class DatasetController : DatasetApi, AuthorizingComponent {
             @PathVariable(TABLE_ID) tableId: UUID,
             @PathVariable(ROW_COUNT) rowCount: Int): Map<UUID, List<Any?>> {
         ensureReadAccess(AclKey(tableId))
-        return edms.getExternalDatabaseTableData(organizationId, tableId, rowCount)
+        val columns = edms.getExternalDatabaseTableWithColumns(tableId).columns
+        val authorizedColumnIds = getAuthorizedColumnIds(tableId, columns.map { it.id }.toSet(), Permission.READ)
+        if (authorizedColumnIds.isEmpty()) {
+            throw ForbiddenException("Missing required permissions to read table data.")
```

Comment:
Just double checking, Is this supposed to throw an exception if none of the columns are accessible or if one or more?
If it's the earlier, I would change the error message to something like "Unable to read data table $tableId. Missing ${Permission.READ} permission on all columns."

Reply:
Throws if none of the columns are accessible. Will update the message!